django/db/models/sql/query.py

```diff
@@ -1471,7 +1511,7 @@ def setup_joins(self, names, opts, alias, can_reuse=None, allow_many=True,
             joins.append(alias)
             if filtered_relation:
                 filtered_relation.path = joins[:]
-        return JoinInfo(final_field, targets, opts, joins, path)
+        return JoinInfo(final_field, targets, opts, joins, path, final_transformer)
```

Comment:
I've added a description of the transform function to the docstring but I didn't include information about the named tuple, perhaps I should have done this as part of #8840 but I think maybe if we think this should reference JoinInfo that can be a small improvement rather than adding to this monster of a ticket?

Reply:
I'd rather see the docstring updated in this one: just so it's not forgotten. With the change, the paragraph beginning `Return...` isn't right. 

My worry is if we leave it then it'll be forgotten.

Reply:
Please don't suggest doing unrelated cleanups along with bug fix or feature work. We try to avoid that as it confuses things when studying a patch.

Reply:
I added the transform function info to the 'Return...' paragraph so that it won't be incorrect, I agree that it's not the clearest docstring in the world an that improving it would be worthwhile, but we already split out the JoinInfo cleanup from this ticket and I don't really want to reintroduce it.

Reply:
Thanks @MatthewWilkes. It was only that sentence/paragraph that needs fixing here, since it's adding the new value that will be unpacked. Any other clean up can wait.